src/enzyme_ad/jax/Passes/AffineToStableHLORaising.cpp

```diff
@@ -944,6 +978,217 @@ emitStoreAsScatter(Location loc, Value update, Value input, ValueRange sIndices,
   return res;
 }
 
+static LogicalResult tryRaisingForOpToStableHLOUnroll(
+    affine::AffineForOp forOp, IRMapping &mapping, OpBuilder &builder,
+    llvm::DenseMap<Value, affine::AffineValueMap> &maps, ParallelContext pc) {
+  auto tmpBlock = new Block();
+  OpBuilder oldFuncBuilder(builder.getContext());
```

Comment:
Nit: prefer updating the insertion point of the existing builder, potentially along with `InsertionGuard`. This makes sure the builder/rewriter configuration and listeners is correctly preserved.

Reply:
I deliberately created a new builder here because the builder we pass in is for building the raised version of the function and the builder here is only used for building a temporary unrolled version of the loop. None of the operations created by the new builder will make it alive out of this function.